wolfCLU/src/crypto/clu_crypto_setup.c

```diff
@@ -0,0 +1,358 @@
+        return MEMORY_E;
+    }
+    XMEMSET(iv, 0, block);
+
+    key = (byte*) XMALLOC(size, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+    if (key == NULL) {
+        wolfCLU_freeBins(pwdKey, iv, NULL, NULL, NULL);
+        return MEMORY_E;
+    }
+    XMEMSET(key, 0, size);
+
+    opterr = 0; /* do not display unrecognized options */
+    optind = 0; /* start at indent 0 */
+    while ((option = getopt_long_only(argc, argv, "",
+                   crypt_options, &long_index )) != -1) {
+
+        switch (option) {
+        case WOLFCLU_PASSWORD: /* Password                                  */
+            XSTRNCPY((char*)pwdKey, optarg, size);
+            pwdKeyChk = 1;
+            keyType   = 1;
+            break;
+
+        case WOLFCLU_PBKDF2:
+            pbkVersion = 2;
+            break;
+
+        case WOLFCLU_KEY:      /* Key if used must be in hex                */
+            break;
+
+        case WOLFCLU_IV:       /* IV if used must be in hex                 */
+            {
+                char ivString[XSTRLEN(optarg)];
+                XSTRNCPY(ivString, optarg, XSTRLEN(optarg));
+                ret = wolfCLU_hexToBin(ivString, &iv, &ivSize,
+                                       NULL, NULL, NULL,
+                                       NULL, NULL, NULL,
+                                       NULL, NULL, NULL);
+                if (ret != 0) {
+                    printf("failed during conversion of IV, ret = %d\n",
+                           ret);
+                    wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                    return -1;
+                }
+                ivCheck = 1;
+            }
+            break;
+
+        case WOLFCLU_SIGN:
+            break;
+
+        case WOLFCLU_VERIFY:   /* Verify results, used with -iv and -key    */
+            /* using hexidecimal format */
+            inputHex = 1;
+            break;
+
+        case WOLFCLU_INFORM:   /* Certificate Stuff                         */
+        case WOLFCLU_OUTFORM:
+        case WOLFCLU_OUTPUT:
+        case WOLFCLU_NOOUT:
+        case WOLFCLU_TEXT_OUT:
+        case WOLFCLU_SILENT:
+        case WOLFCLU_PUBIN:
+        case WOLFCLU_PUBOUT:
+        case WOLFCLU_PUBKEY:
+
+
+            /* The cases above have their arguments converted to lower case */
+            if (optarg) convert_to_lower(optarg, (int)XSTRLEN(optarg));
+            /* The cases below won't have their argument's molested */
+
+        case WOLFCLU_INFILE:   /* File passed in by user                    */
+            in = optarg;
+            inCheck = 1;
+            break;
+
+        case WOLFCLU_OUTFILE:  /* Output file                               */
+            out = optarg;
+            outCheck = 1;
+            break;
+
+        case WOLFCLU_INKEY:
+            if (optarg == NULL) {
+                printf("no key passed in..\n");
+                wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                return FATAL_ERROR;
```

Comment:
Earlier we were returning -1 for errors. Why `FATAL_ERROR` now? Can we standardize on one? I would prefer the latter, since it conveys more meaning than -1.